docs/proposals/autoscaling.md

```diff
@@ -0,0 +1,245 @@
+        
+        //MonitorSelector defines a set of capacity that the auto-scaler is monitoring (replication controllers).  Generally, the auto-scaler is 
+        //driven by the AutoScaleThresholds, however, this gives visibility of aggregate items like total number of pods
+        //backing a service without having to aggregate them into a statistic
+        MonitorSelector map[string]string
+    }
+     
+    type AutoScalerStatus struct {
+        // TODO: open for discussion on what meaningful information can be reported in the status
+        // The status may return the replica count here but we may want more information
+        // such as if the count reflects a threshold being passed
+    }     
+     
+     
+     //AutoScaleThresholdInterface abstracts the data analysis from the auto-scaler
+     //example: scale by 1 (Increment) when RequestsPerSecond (Type) pass comparison (Comparison) of 50 (Value) for 30 seconds (Duration)
+     type AutoScaleThresholdInterface interface {
+        //called by the auto-scaler to determine if this threshold is met or not
+        ShouldScale() boolean
+     }
+      
+        
+     //AutoScaleThreshold is a single statistic used to drive the auto-scaler in scaling decisions
+     type AutoScaleThreshold struct {
+        // Type is the type of threshold being used, intention or value
+        Type AutoScaleThresholdType
+        
+        // ValueConfig holds the config for value based thresholds
+        ValueConfig AutoScaleValueThresholdConfig
+        
+        // IntentionConfig holds the config for intention based thresholds
+        IntentionConfig AutoScaleIntentionThresholdConfig                
+     } 
+     
+     // AutoScaleIntentionThresholdConfig holds configuration for intention based thresholds
+     // a intention based threshold defines no increment, the scaler will adjust by 1 accordingly and 
+     // maintain once the intention is reached.  Also, no selector is defined, the intention should
+     // dictate the selector used for statistics.  Same for duration although we may want a configurable
+     // duration later so intentions are more customizable.
+     type AutoScaleIntentionThresholdConfig struct {
+        // Intent is the lexicon of what intention is requested
+        Intent  AutoScaleIntentionType
+        
+        // Value is intention dependent in terms of above, below, equal and represents the value to check against
+        Value   float
+     }
+     
+     // AutoScaleValueThresholdConfig holds configuration for value based thresholds
+     type AutoScaleValueThresholdConfig struct {
+        //Increment determines how the auot-scaler should scale up or down (positive number to scale up based on this threshold
+        //negative number to scale down by this threshold)
+        Increment int
+        //Selector is the statistics selector that determines how the Threshold receives data from aggregated (or single) statistics. 
```

Comment:
Sorry, missed addressing this one.  

The selector is meant to represent a way to query the data from whatever storage facility is used for statistics.  Once the storage of statistics are better defined I think the retrieval mechanism will become more obvious.  I will update the proposal so it is more obvious what the intention is.  

The actual type that is selected would be something that represents an individual statistic which may or may not be an aggregation of other statistics.  The statistic should boil down to a single value that can be compared to the threshold limits.